servicetalk-dns-discovery-netty/src/test/java/io/servicetalk/dns/discovery/netty/DefaultDnsClientTest.java

```diff
@@ -984,6 +984,25 @@ void srvExceptionInSubscriberOnNext() throws Exception {
         latchOnError.await();
     }
 
+    @Test
+    void capsMaxTTL() throws Exception {
+        setup(builder -> builder.maxTTL(3));
+        final String domain = "servicetalk.io";
+        String ip = nextIp();
+        recordStore.addIPv4Address(domain, 5, ip);
+
+        TestPublisherSubscriber<ServiceDiscovererEvent<InetAddress>> subscriber = dnsQuery(domain);
+        Subscription subscription = subscriber.awaitSubscription();
+        subscription.request(Long.MAX_VALUE);
+
+        System.err.println(subscriber.takeOnNext());
+        recordStore.removeIPv4Address(domain, 1, ip);
+        advanceTime(3);
+        System.err.println(subscriber.takeOnNext());
+
+        //assertEvent(subscriber.takeOnNext(), ipv6, AVAILABLE);
```

Comment:
Discussed offline: we have to add a new IP in order for events to propagate.
Please also add a test for SRV records.